src/coreclr/vm/method.hpp

```diff
@@ -1813,22 +1822,25 @@ class MethodDesc
         m_wFlags |= mdfHasNativeCodeSlot;
     }
 
-    inline bool IsAsyncThunkMethod()
+    inline bool IsAsyncHelperMethod() const
     {
         LIMITED_METHOD_DAC_CONTRACT;
         if (!HasAsyncMethodData())
             return false;
-        AsyncMethodType asyncType = GetAddrOfAsyncMethodData()->type;
-        return asyncType == AsyncMethodType::AsyncToTask || asyncType == AsyncMethodType::TaskToAsync;
+        auto asyncType = GetAddrOfAsyncMethodData()->type;
```

Comment:
```suggestion
        AsyncMethodKind asyncType = GetAddrOfAsyncMethodData()->type;
```
and potentially a further rename of the variable.